### PR TITLE
Appfeature app

### PR DESCRIPTION
**Pull request by tnyandoro**
>We created a script file enumerable.rb to house methods.
>We added the methods onto the existing enumerable module
> We created a #my_each method that is identical to #each
> We created a #my_ieach_index method in the same way
> We created  #my_all? (continue as above)
> We created   #my_any?
>We created   #my_none?
>We created  #my_count
>We created   #my_map
>We created   #my_inject
> We tested our #my_inject by creating a method called #multiply_els which multiplies all the elements of the array together by using #my_inject, e.g. multiply_els([2,4,5]) #=> 40
> We Modified our #my_map method to take a proc instead

### CODE REVIEW

**Review by andresporras3423 — CHANGES_REQUESTED** [review]
### Status: Changes Required :recycle:
**Hello Tendai Nyandoro** 
Nice job with your first submission for this project :muscle:  :muscle:
Your project passes most of the tests :clap: :clap: :clap:  
There are some issues to correct before you can go ahead:

- [ ] fix rubocop offenses
- [ ] #my_each when a block is given when self is a range returns the range itself
after calling the given block once for each element in self
        `[1,2,3].my_each(&proc{|x| x>2}) == [1,2,3]`
- [ ] #my_each_with_index returns an enumerator if no block is given
    `[1,2,3].my_each_with_index.class == Enumerator`
- [ ] #my_select returns an enumerator if no block is given
    `[1,2,3,4].my_select.class==Enumerator`
- [ ] #my_all? when no block or argument is given returns true when none of the collection members are false or nil
      `[true, [false]].all? == [true, [false]].my_all?`
- [ ] #my_all? when no block or argument is given returns false when one of the collection members are false or nil
      `[true,[true],false].all? == [true,[true],false].my_all?`
- [ ] #my_none? returns true if the block never returns true for all elements
    `[1,2,3,4,5].my_none?(&proc{|n| n%7==0}) == [1,2,3,4,5].none?(&proc{|n| n%7==0})`
- [ ] #my_none? when no block or argument is given returns true only if none of the collection members is true
      `[false, nil,false].none? == [false, nil,false].my_none?`
- [ ] #my_none? when a Regex is passed as an argument returns true if none of the
collection matches the Regex
      `['dog','cat'].none?(/x/) ==  ['dog','cat'].my_none?(/x/)`
- [ ] #my_none? when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
      `['dog','car'].none?(5) == ['dog','car'].my_none?(5)`
- [ ] #my_count counts the number of items in enum that are equal to item if an argument is given
    `[1,2,3].count(1) == [1,2,3].my_count(1)`
- [ ] #my_count counts the number of elements yielding a true value if a block is
given
    `[1,2,3].count(&proc{|num|num%2==0}) == [1,2,3].my_count(&proc{|num|num%2==0})`
- [ ] #my_inject raises a 'LocalJumpError' when no block or argument is given
    ```
    begin
      [1,2,3,4].my_inject
    rescue LocalJumpError
      true
    end
    ```

 
Please request another review once you are ready! :+1: 
 
Happy coding :relaxed:
 
Oscar Andrés Russi Porras

**Comment by tnyandoro** [comment]
Change request fixed

File: enumerable.rb [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    while i < to_a.length
+      yield to_a([i], i)
+      i += 1
+    end
+    self
+  end
```
Comment (maelfosso) [inline enumerable.rb]:
```rb
2.6.5 :011 > block = proc { |num| num < 4 }
2.6.5 :012 > array.my_each_with_index(&block)
Traceback (most recent call last):
        8: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/ruby_executable_hooks:24:in `<main>'
        7: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/ruby_executable_hooks:24:in `eval'
        6: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/irb:23:in `<main>'
        5: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/irb:23:in `load'
        4: from /home/maelfosso/.rvm/gems/ruby-2.6.5/gems/irb-1.2.4/exe/irb:11:in `<top (required)>'
        3: from (irb):12
        2: from enumerable.rb:19:in `my_each_with_index'
        1: from enumerable.rb:19:in `to_a'
ArgumentError (wrong number of arguments (given 2, expected 0))
2.6.5 :013 > array.each_with_index(&block)
 => [5, 2, 4, 4, 1, 4, 8, 2, 1, 0, 2, 7, 6, 6, 0, 7, 5, 4, 4, 5, 0, 7, 3, 6, 7, 0, 7, 8, 1, 8, 3, 3, 5, 6, 7, 0, 0, 2, 0, 7, 8, 6, 7, 7, 7, 7, 0, 6, 1, 8, 6, 8, 8, 2, 3, 5, 5, 0, 0, 7, 6, 7, 8, 5, 1, 3, 7, 4, 4, 8, 2, 7, 7, 6, 8, 8, 3, 6, 0, 2, 8, 4, 1, 1, 7, 1, 3, 1, 2, 3, 6, 3, 1, 3, 0, 3, 4, 4, 6, 8] 
2.6.5 :014 > 
2.6.5 :015 > array
 => [5, 2, 4, 4, 1, 4, 8, 2, 1, 0, 2, 7, 6, 6, 0, 7, 5, 4, 4, 5, 0, 7, 3, 6, 7, 0, 7, 8, 1, 8, 3, 3, 5, 6, 7, 0, 0, 2, 0, 7, 8, 6, 7, 7, 7, 7, 0, 6, 1, 8, 6, 8, 8, 2, 3, 5, 5, 0, 0, 7, 6, 7, 8, 5, 1, 3, 7, 4, 4, 8, 2, 7, 7, 6, 8, 8, 3, 6, 0, 2, 8, 4, 1, 1, 7, 1, 3, 1, 2, 3, 6, 3, 1, 3, 0, 3, 4, 4, 6, 8] 
2.6.5 :016 > 

```

Comment (maelfosso) [reply enumerable.rb]:
```rb
2.6.5 :018 > range = Range.new(5, 50)
2.6.5 :019 > range.each_with_index(&block)
 => 5..50 
2.6.5 :020 > range.my_each_with_index(&block)
Traceback (most recent call last):
        8: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/ruby_executable_hooks:24:in `<main>'
        7: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/ruby_executable_hooks:24:in `eval'
        6: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/irb:23:in `<main>'
        5: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/irb:23:in `load'
        4: from /home/maelfosso/.rvm/gems/ruby-2.6.5/gems/irb-1.2.4/exe/irb:11:in `<top (required)>'
        3: from (irb):20
        2: from enumerable.rb:19:in `my_each_with_index'
        1: from enumerable.rb:19:in `to_a'
ArgumentError (wrong number of arguments (given 2, expected 0))
2.6.5 :021 >
```

Comment (maelfosso) [reply enumerable.rb]:
```rb
2.6.5 :023 > array
 => [5, 2, 4, 4, 1, 4, 8, 2, 1, 0, 2, 7, 6, 6, 0, 7, 5, 4, 4, 5, 0, 7, 3, 6, 7, 0, 7, 8, 1, 8, 3, 3, 5, 6, 7, 0, 0, 2, 0, 7, 8, 6, 7, 7, 7, 7, 0, 6, 1, 8, 6, 8, 8, 2, 3, 5, 5, 0, 0, 7, 6, 7, 8, 5, 1, 3, 7, 4, 4, 8, 2, 7, 7, 6, 8, 8, 3, 6, 0, 2, 8, 4, 1, 1, 7, 1, 3, 1, 2, 3, 6, 3, 1, 3, 0, 3, 4, 4, 6, 8] 
2.6.5 :024 > array_clone = array.dup
2.6.5 :025 > array.my_each_with_index { |num| num + 1 }
Traceback (most recent call last):
        8: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/ruby_executable_hooks:24:in `<main>'
        7: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/ruby_executable_hooks:24:in `eval'
        6: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/irb:23:in `<main>'
        5: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/irb:23:in `load'
        4: from /home/maelfosso/.rvm/gems/ruby-2.6.5/gems/irb-1.2.4/exe/irb:11:in `<top (required)>'
        3: from (irb):25
        2: from enumerable.rb:19:in `my_each_with_index'
        1: from enumerable.rb:19:in `to_a'
ArgumentError (wrong number of arguments (given 2, expected 0))
2.6.5 :026 >
```

Comment (maelfosso) [reply enumerable.rb]:
```rb
2.6.5 :028 > my_each_with_index_output = ''
2.6.5 :029 > block = proc { |num, idx| my_each_with_index_output += "Num: #{num}, idx: #{idx}\n" }
2.6.5 :030 > hash = { a: 1, b: 2, c: 3, d: 4, e: 5 }
2.6.5 :031 > hash.each_with_index(&block)
 => {:a=>1, :b=>2, :c=>3, :d=>4, :e=>5} 
2.6.5 :032 > each_with_index_output = my_each_with_index_output.dup
2.6.5 :033 > my_each_with_index_output = ''
2.6.5 :034 > hash.my_each_with_index(&block)
Traceback (most recent call last):
        8: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/ruby_executable_hooks:24:in `<main>'
        7: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/ruby_executable_hooks:24:in `eval'
        6: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/irb:23:in `<main>'
        5: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/irb:23:in `load'
        4: from /home/maelfosso/.rvm/gems/ruby-2.6.5/gems/irb-1.2.4/exe/irb:11:in `<top (required)>'
        3: from (irb):34
        2: from enumerable.rb:19:in `my_each_with_index'
        1: from enumerable.rb:19:in `to_a'
ArgumentError (wrong number of arguments (given 2, expected 0))
2.6.5 :035 > 
```

File: enumerable.rb [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+        end
+      else
+        to_a.my.each do |i|
+          status = !(i == false || i.nil?)
+          break unless status
+        end
+      end
+    end
+    status
+  end
```
Comment (maelfosso) [inline enumerable.rb]:
```rb
2.6.5 :037 > true_array = [1, true, 'hi', []]
2.6.5 :038 > true_array.all?
 => true 
2.6.5 :039 > true_array.my_all?
 => #<Enumerator: [1, true, "hi", []]:my_all?> 
2.6.5 :040 > 
```

Comment (maelfosso) [reply enumerable.rb]:
```rb
2.6.5 :042 > false_array = [1, false, 'hi', []]
2.6.5 :043 > false_array.all?
 => false 
2.6.5 :044 > false_array.my_all?
 => #<Enumerator: [1, false, "hi", []]:my_all?> 
2.6.5 :045 > 
```

Comment (maelfosso) [reply enumerable.rb]:
```rb
2.6.5 :047 > array
 => [5, 2, 4, 4, 1, 4, 8, 2, 1, 0, 2, 7, 6, 6, 0, 7, 5, 4, 4, 5, 0, 7, 3, 6, 7, 0, 7, 8, 1, 8, 3, 3, 5, 6, 7, 0, 0, 2, 0, 7, 8, 6, 7, 7, 7, 7, 0, 6, 1, 8, 6, 8, 8, 2, 3, 5, 5, 0, 0, 7, 6, 7, 8, 5, 1, 3, 7, 4, 4, 8, 2, 7, 7, 6, 8, 8, 3, 6, 0, 2, 8, 4, 1, 1, 7, 1, 3, 1, 2, 3, 6, 3, 1, 3, 0, 3, 4, 4, 6, 8] 
2.6.5 :048 > array.all?(Integer)
 => true 
2.6.5 :049 > array.my_all?(Integer)
 => #<Enumerator: [5, 2, 4, 4, 1, 4, 8, 2, 1, 0, 2, 7, 6, 6, 0, 7, 5, 4, 4, 5, 0, 7, 3, 6, 7, 0, 7, 8, 1, 8, 3, 3, 5, 6, 7, 0, 0, 2, 0, 7, 8, 6, 7, 7, 7, 7, 0, 6, 1, 8, 6, 8, 8, 2, 3, 5, 5, 0, 0, 7, 6, 7, 8, 5, 1, 3, 7, 4, 4, 8, 2, 7, 7, 6, 8, 8, 3, 6, 0, 2, 8, 4, 1, 1, 7, 1, 3, 1, 2, 3, 6, 3, 1, 3, 0, 3, 4, 4, 6, 8]:my_all?> 
2.6.5 :050 > 
```

Comment (maelfosso) [reply enumerable.rb]:
```rb
2.6.5 :052 > numbers = [1, 2i, 3.14]
2.6.5 :053 > numbers.my_all?(Numeric)
 => #<Enumerator: [1, (0+2i), 3.14]:my_all?> 
2.6.5 :054 > numbers.all?(Numeric)
 => true 
2.6.5 :055 > 
```

Comment (maelfosso) [reply enumerable.rb]:
```rb
2.6.5 :057 > array
 => [5, 2, 4, 4, 1, 4, 8, 2, 1, 0, 2, 7, 6, 6, 0, 7, 5, 4, 4, 5, 0, 7, 3, 6, 7, 0, 7, 8, 1, 8, 3, 3, 5, 6, 7, 0, 0, 2, 0, 7, 8, 6, 7, 7, 7, 7, 0, 6, 1, 8, 6, 8, 8, 2, 3, 5, 5, 0, 0, 7, 6, 7, 8, 5, 1, 3, 7, 4, 4, 8, 2, 7, 7, 6, 8, 8, 3, 6, 0, 2, 8, 4, 1, 1, 7, 1, 3, 1, 2, 3, 6, 3, 1, 3, 0, 3, 4, 4, 6, 8] 
2.6.5 :058 > array.my_all?(Integer)
 => #<Enumerator: [5, 2, 4, 4, 1, 4, 8, 2, 1, 0, 2, 7, 6, 6, 0, 7, 5, 4, 4, 5, 0, 7, 3, 6, 7, 0, 7, 8, 1, 8, 3, 3, 5, 6, 7, 0, 0, 2, 0, 7, 8, 6, 7, 7, 7, 7, 0, 6, 1, 8, 6, 8, 8, 2, 3, 5, 5, 0, 0, 7, 6, 7, 8, 5, 1, 3, 7, 4, 4, 8, 2, 7, 7, 6, 8, 8, 3, 6, 0, 2, 8, 4, 1, 1, 7, 1, 3, 1, 2, 3, 6, 3, 1, 3, 0, 3, 4, 4, 6, 8]:my_all?> 
2.6.5 :059 > array.all?(Integer)
 => true 
2.6.5 :060 > 
```

Comment (maelfosso) [reply enumerable.rb]:
```rb
2.6.5 :062 > words = %w[dog door rod blade]
2.6.5 :063 > words.my_all?(/o/)
 => #<Enumerator: ["dog", "door", "rod", "blade"]:my_all?> 
2.6.5 :064 > words.all?(/o/)
 => false 
2.6.5 :065 >
```

Comment (maelfosso) [reply enumerable.rb]:
```rb
2.6.5 :066 > array
 => [5, 2, 4, 4, 1, 4, 8, 2, 1, 0, 2, 7, 6, 6, 0, 7, 5, 4, 4, 5, 0, 7, 3, 6, 7, 0, 7, 8, 1, 8, 3, 3, 5, 6, 7, 0, 0, 2, 0, 7, 8, 6, 7, 7, 7, 7, 0, 6, 1, 8, 6, 8, 8, 2, 3, 5, 5, 0, 0, 7, 6, 7, 8, 5, 1, 3, 7, 4, 4, 8, 2, 7, 7, 6, 8, 8, 3, 6, 0, 2, 8, 4, 1, 1, 7, 1, 3, 1, 2, 3, 6, 3, 1, 3, 0, 3, 4, 4, 6, 8] 
2.6.5 :067 > array.my_all?(3)
 => #<Enumerator: [5, 2, 4, 4, 1, 4, 8, 2, 1, 0, 2, 7, 6, 6, 0, 7, 5, 4, 4, 5, 0, 7, 3, 6, 7, 0, 7, 8, 1, 8, 3, 3, 5, 6, 7, 0, 0, 2, 0, 7, 8, 6, 7, 7, 7, 7, 0, 6, 1, 8, 6, 8, 8, 2, 3, 5, 5, 0, 0, 7, 6, 7, 8, 5, 1, 3, 7, 4, 4, 8, 2, 7, 7, 6, 8, 8, 3, 6, 0, 2, 8, 4, 1, 1, 7, 1, 3, 1, 2, 3, 6, 3, 1, 3, 0, 3, 4, 4, 6, 8]:my_all?> 
2.6.5 :068 > array.all?(3)
 => false 
2.6.5 :069 > 
```

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required :red_circle: 

Hello,

Thank you for implementing the suggested changes
However, there are some issues you must solve

Please, read the comments they will guide you
Once changes are done, submit a new code review request

Thanks
Mael FOSSO

**Review by teekaytech — CHANGES_REQUESTED** [review]
## Status: Changes Required :warning: :vertical_traffic_light: 

### Hello team :wave:

__A lot has been done in this project and I commend your effort so far. However, we still have some failed test cases. Please check my points below and fix them as appropriate.__

**[ REQUIRED CHANGES ]**

- [ ] **You seem to have made some changes to your `.rubocop.yml` linter file.** 
- You should not do this. Instead, you can disable some rubocop rules. Check out [this article](https://rubydoc.info/gems/rubocop/0.24.1/frames#Disabling_Cops_within_Source_Code) on how to do that. 
- Please change your `.rubocop.yml` file content to the default one provided by [microverse](https://github.com/microverseinc/linters-config/blob/master/ruby/.rubocop.yml) 
- After that, kindly fix all linter errors before you submit another review request.



- [ ] You have almost 15 failed test cases. Because of my time, I cannot explain deeply most of these cases. But I will try to summarize things as much as I can. Feel free to reach out to any of the previous TSEs or me if you are not clear about anything.

**_Please check [this](https://ruby-doc.org/core-2.6.5/Enumerable.html) documentation on enumerables for better understanding._**

- **`my_each_with_index` failed 6 test cases.** All these failed cases have the same reason for failing, which is, you did not define your `my_each_with_index` with its two(2) arguments. The `each_with_index` calls block with two arguments, the item, and its index, for each item in the enum/array. Try to take care of these arguments in your implementation. For instance, 

```
hash = Hash.new
%w(cat dog wombat).each_with_index { |item, index|
  hash[item] = index
}
p hash # returns {"cat"=>0, "dog"=>1, "wombat"=>2}

%w(cat dog wombat).my_each_with_index { |item, index|
  hash[item] = index
}  #returns error: wrong number of arguments (given 2, expected 0) (ArgumentError)
```


- **Your `my_all?` failed 9 test cases.** Please review your implementation of the `my_all` enumerable method so that it takes care of different arguments. [This docs](https://apidock.com/ruby/v2_5_5/Enumerable/all%3F) might help. Here are some of the conditions test case instances.

- When no block or argument is given, the `my_all?` should return true when none of the collection members are false or nil. E.g.

```
p [1, true, 'hi', []].all? # returns true
p [1, true, 'hi', []].my_all? # returns #<Enumerator: [1, true, "hi", []]:my_all?>
```

- When a class is passed as an argument, `my_all?` should return true if all of the collection is a member of such (e.g. class:: Integer) E.g.

```
p [1,2,3,4].all?(Integer) # returns true
p [1,2,3,4].my_all?(Integer) # returns #<Enumerator: [1, 2, 3, 4]:my_all?>
```

-  When a Regex is passed as an argument, `my_all?` should return true if all of the collection matches the Regex. e.g.

```
p %w[dog door rod blade].all?(/d/) # returns true
p %w[dog door rod blade].my_all?(/d/) # returns #<Enumerator: ["dog", "door", "rod", "blade"]:my_all?>
```

- When a pattern other than Regex or a Class is given, `my_all? should return true if all of the collection matches the pattern. e.g.

```
p [1,2,3,4].all?(3) # returns true
p [1,2,3,4].my_all?(3) # returns #<Enumerator: [1, 2, 3, 4]:my_all?>
```

* * * 
**Please make the required changes and submit another review request when you are done.**

* * *

**_You can reach out to me on Twitter (@ola_lere), Github(@teekaytech), or slack (@taofeekolalere )_**
__Happy Coding :smiley:__

**Comment by tnyandoro** [comment]
>All the changes have been made
>Tests have passed

File: enumerable.rb [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+    pos = 0
+    my_arr = []
+    while pos < arr.length
+      yield(arr[pos], pos)
+      my_arr << arr[pos]
+      my_arr << pos
+      pos += 1
+    end
+    my_arr
+  end
```
Comment (SpaYco) [inline enumerable.rb]: - [ ] when a block is given when self is an array it should return the array itself after calling the given block once for each element in self

```Ruby
  array = Array.new(100) { rand(0...9) }
  block = proc { |num| num < (0+ 9) / 2 }

  array.my_each_with_index(&block)
  # should be [5, 7, 1, 4, 1, 2, 0, 7, 7, 5, 5, 5, 0, 4, 0, 6, 6, 1, 6, 6, 4, 4, 3, 2, 4, 2, 3, 7, 0, 1, 4, 2, 8, 2..., 5, 4, 8, 1, 4, 6, 3, 8, 7, 2, 8, 2, 3, 3, 2, 6, 1, 7, 6, 6, 8, 2, 4, 2, 5, 1, 5, 0, 4, 6, 7, 3, 7]
  # but got [5, 0, 7, 1, 1, 2, 4, 3, 1, 4, 2, 5, 0, 6, 7, 7, 7, 8, 5, 9, 5, 10, 5, 11, 0, 12, 4, 13, 0, 14, 6, 15...5, 6, 86, 8, 87, 2, 88, 4, 89, 2, 90, 5, 91, 1, 92, 5, 93, 0, 94, 4, 95, 6, 96, 7, 97, 3, 98, 7, 99] instead

```

- [ ] when a block is given when self is a range it should return the range itself after calling the given block once for each element in self


```Ruby
  block = proc { |num| num < (0+ 9) / 2 }
  range = Range.new(5, 50)

  range.my_each_with_index(&block)
  # should be 5..50
  # but got [5, 0, 6, 1, 7, 2, 8, 3, 9, 4, 10, 5, 11, 6, 12, 7, 13, 8, 14, 9, 15, 10, 16, 11, 17, 12, 18, 13, 19,... 33, 39, 34, 40, 35, 41, 36, 42, 37, 43, 38, 44, 39, 45, 40, 46, 41, 47, 42, 48, 43, 49, 44, 50, 45] instead

```

**Review by SpaYco — CHANGES_REQUESTED** [review]
# Changes Requested 

Hello Team! 👋 
Awesome job so far!! 👍 you still need to make some changes though! you're only 3 fixes away from approval

## Required Changes ♻️

- [ ]  Do not make any changes in config files - they represent style guidelines that you share with your team - which is a group of all Microverse students. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/linter_config.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/54883773/)) (`Style/CaseEquality` can be disabled)

_**Note: in this project, some cops like `Metrics/CyclomaticComplexity` will be needed to be disabled and too difficult to solve, you can disable them using [this method](https://docs.rubocop.org/rubocop/configuration.html#disabling-cops-within-source-code) (the Disabling Cops within Source Code section) instead of changing the config files.**_

- [ ] check my comments below 👇 

### please make the appropriate changes and ask for another code review ✔️ 

Happy Coding!

Best, 
Aziz Mejri 
_contact me on [twitter](https://twitter.com/iSpaYco) or slack (@SpaYco) if you need any help_

File: enumerable.rb (hunk 3 above)
Comment (tnyandoro) [reply enumerable.rb]: ## Please kindly clarify
 ( array = Array.new(100) { rand(0...9) } Random number generated so they will differ
  block = proc { |num| num < (0+ 9) / 2 }

  array.my_each_with_index(&block)
   should be [5, 7, 1, 4, 1, 2, 0, 7, 7, 5, 5, 5, 0, 4, 0, 6, 6, 1, 6, 6, 4, 4, 3, 2, 4, 2, 3, 7, 0, 1, 4, 2, 8, 2..., 5, 4, 8, 1, 4, 6, 3, 8, 7, 2, 8, 2, 3, 3, 2, 6, 1, 7, 6, 6, 8, 2, 4, 2, 5, 1, 5, 0, 4, 6, 7, 3, 7]
   but got [5, 0, 7, 1, 1, 2, 4, 3, 1, 4, 2, 5, 0, 6, 7, 7)

Comment (SpaYco) [reply enumerable.rb]: hey, @tnyandoro, we compare the result using #my_each_with_index and #each_with_index using the same variables, so even if the number is randomly generated, it should give the same result :)

**Review by SpaYco — CHANGES_REQUESTED** [review]
## Changes Requested

Hello again 👋, since you haven't committed any change, I will mark this project again as 'changes requested', please make the changes using my previous review and request again 

Best, 
Aziz Mejri 
_contact me on [twitter](https://twitter.com/iSpaYco) or slack (@SpaYco) if you need any help_

**Review by kevcap — APPROVED** [review]
# [STATUS] PROJECT APPROVED 🍾

### I really like your project! Congrats team, you guys did an excellent job, you have my approval to go ahead and merge this pull request!

_**Clean and modular code, following the best practices, keep it up!**_ 🥇

**Disclaimer: Sorry for the delay, I was off from microverse since Wednesday, but somehow your project was assigned to me, and I just get noticed 5min ago.**


## As always, I wish you all the very best and see you both at the top🚀


👨🏻‍💻 **TSE: Kevin Alves**

  

📫 Have any question? Better call Kevin:

<a  href="mailto:kevinalves.js@gmail.com"> gmail</a> [Twitter Follow](https://twitter.com/intent/follow?screen_name=kevinnnmn) [](https://microverse-students.slack.com/team/UT86D3AP4)

**Pull request merged**